operator/internal/manifests/storage/schema.go

```diff
@@ -0,0 +1,200 @@
+package storage
+
+import (
+	"sort"
+	"time"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	lokiv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+)
+
+const (
+	// DateTimeFormat is the datetime string need to format the time.
+	DateTimeFormat = "2006-01-02"
+
+	// UpdateDelay is the time before a change to the storage schema
+	// is applied to a cluster. This will avoid scenarios in which
+	// the difference between UTC and local time zones will cause
+	// logs to be unavailable.
+	UpdateDelay = time.Hour * 24
```

Comment:
The reason for the two day delay in the hit is to make sure we clear past the UTC/Local midnight scenario. I'm not sure if just a few hours of time will clear the time in some region. However, I don't know the max value of difference there can be between the two. However, the two day window this provides should clear the differences.

(It's two days because the anything within a day of the current time is counted as an applied change. Since the schema only works in year-month-day formats, the delay is actually two days.)

Reply:
It's not that I mean there is a delay in the configuration. What I'm concerned about allowing is the given example: https://grafana.com/docs/loki/latest/operations/storage/schema/#changing-the-schema

Granted, this part of the overhead should be assumed by the person adding the schema. However, the enforcing of this delay should at least make this scenario hard to cause. Though I agree that the timing capture here is not obvious. What I'm trying to avoid is someone telling Loki, make a change tomorrow (local time) and it is already tomorrow for Loki (utc time). Adding the delay should stop anyone from making the `effectiveDate` either the current date.

I guess it should be stated here that all of the dates entered on the spec are assumed to be in UTC. While I could do some assumptions on the location via `time.Now().Location()`, that may have areas where it might not work out. Assuming that the location could be trusted, I suppose doing a validation check against that time converted to UTC would be the most ideal way to stop this from occurring.

Then in that case, we'd probably only need to budget in an hour to avoid any surprise midnight change.